src/Microsoft.Extensions.Logging.Console/Internal/ConsoleLoggerProcessor.cs

```diff
@@ -27,7 +27,10 @@ public ConsoleLoggerProcessor()
 
         public virtual void EnqueueMessage(LogMessageEntry message)
         {
-            _messageQueue.Add(message);
+            if (!_messageQueue.IsAddingCompleted)
```

Comment:
Other threads can call `EnqueueMessage` while Dispose is being called, so this doesn't get rid of the issue, just decreases how often it happens.